snappy/snap_remote_repo.go

```diff
@@ -230,10 +229,20 @@ func (s *SnapUbuntuStoreRepository) Details(name string, origin string) (parts [
 		return nil, err
 	}
 
-	snap := NewRemoteSnapPart(detailsData)
-	parts = append(parts, snap)
+	return NewRemoteSnapPart(detailsData), nil
+}
 
-	return parts, nil
+// Details returns details for the given snap in this repository
+func (s *SnapUbuntuStoreRepository) Details(name string, origin string) (parts []Part, err error) {
+	snapName := name
```

Comment:
Indeed, much agreed. The API for Details() is also terrible. It only exists because not all code was converted yet. I fixed all of snappy/ now, the daemon is missing and I will do that in a followup branch (its a bit more work because it uses heavy mocking so the diff gets bigger).